src/postamble.js

```diff
@@ -150,11 +150,15 @@ function callMain(args) {
 
   var argc = args.length+1;
   var argv = stackAlloc((argc + 1) * {{{ Runtime.POINTER_SIZE }}});
-  HEAP32[argv >> 2] = allocateUTF8OnStack(thisProgram);
+  var argv_pos = argv;
+  {{{ makeSetValue('argv_pos', 0, 'allocateUTF8OnStack(thisProgram)', POINTER_TYPE) }}};
+  argv_pos += {{{ Runtime.POINTER_SIZE }}};
   for (var i = 1; i < argc; i++) {
-    HEAP32[(argv >> 2) + i] = allocateUTF8OnStack(args[i - 1]);
+    {{{ makeSetValue('argv_pos', 0, 'allocateUTF8OnStack(args[i - 1])', POINTER_TYPE) }}};
+    argv_pos += {{{ Runtime.POINTER_SIZE }}};
```

Comment:
This will increase code size slightly. Perhaps we could add a helper that does this common pattern, that is,
```js
{{{ makeIndexedSetValue('argv_pos', 'i', 'allocateUTF8OnStack(args[i - 1])', POINTER_TYPE) }}}
```
`makeIndexedSetValue` would treat the second param as an index of the pointer size, so it would emit `HEAP32[(argv_pos >> 2) + i]` or `HEAP64[(argv_pos >> 3) + i]` (in both cases, the `+ i` is the same).

Just a few bytes here, but maybe there are more such locations, and this pattern seems common even if we don't optimize it in all places like here.

But this could also be a TODO

Reply:
Hmm.. I think I can do even better.